coordinator/src/lib.rs

```diff
@@ -116,31 +116,32 @@ impl Coordinator {
     }
 
     fn new_session(&self, storage: &mut dyn StorageAccess) -> SessionId {
-        let mut sessions = self.sessions.write();
-        let (index, bit) = sessions
-            .iter()
-            .enumerate()
-            .find_map(|(i, &bits)| {
-                if bits == SessionSlot::MAX {
-                    None
-                } else {
-                    Some((i, bits.trailing_ones()))
-                }
-            })
-            .unwrap_or_else(|| {
-                sessions.push(0);
-                (sessions.len() - 1, 0)
-            });
+        let session_id = {
+            let mut sessions = self.sessions.write();
+            let (index, bit) = sessions
+                .iter()
+                .enumerate()
+                .find_map(|(i, &bits)| {
+                    if bits == SessionSlot::MAX {
+                        None
+                    } else {
+                        Some((i, bits.trailing_ones()))
+                    }
+                })
+                .unwrap_or_else(|| {
+                    sessions.push(0);
+                    (sessions.len() - 1, 0)
+                });
 
-        sessions[index] |= 1 << bit;
-        let session_id = bit + (SESSION_BITS_PER_SLOT * index) as SessionId;
+            sessions[index] |= 1 << bit;
+            bit + (SESSION_BITS_PER_SLOT * index) as SessionId
+        };
```

Comment:
@junha1 Could you explain how session is using bitset?

Reply:
@dynaxis authored this.

Reply:
It's a bit more complex than absolutely required. I want to assign the smallest integer ID available and manage unoccupied IDs by mapping them to 0 bits (naturally 1 bits for occupied IDs). The code can be implemented using `bool` vector instead.